lib/Reader/reader.cpp

```diff
@@ -2279,7 +2279,7 @@ EHRegion *ReaderBase::fgSwitchRegion(EHRegion *OldRegion, uint32_t Offset,
 
 #define CHECKTARGET(TargetOffset, BufSize)                                     \
   {                                                                            \
-    if ((TargetOffset) < 0 || (TargetOffset) >= (BufSize))                     \
+    if ((TargetOffset) >= (BufSize))                                           \
```

Comment:
It appears both parameters are expected to be unsigned for this negative check to be useless.
Can we turn CHECKTARGET to a function ```checkTarget(uint32_t TargetOffset, uint32_t BufSize)``` to ensure we check unsigned offsets. Or just inline it into each call site since it's literally a simple check.